File: Backend/src/main/java/com/bachld/project/backend/dto/response/donhoandoan/DonHoanDoAnResponse.java
package com.bachld.project.backend.dto.response.donhoandoan;

import com.bachld.project.backend.enums.HoanState;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.LocalDateTime;

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class DonHoanDoAnResponse {
    Long id;

    Long sinhVienId;
    Long deTaiId;

    HoanState trangThai;
    String lyDo;
    String minhChungUrl;

    LocalDateTime requestedAt;
    LocalDateTime decidedAt;

    Long nguoiPheDuyetId;
    String ghiChuQuyetDinh;
}

File: Android/app/src/main/java/com/bachld/android/data/dto/response/donhoandoan/DonHoanDoAnResponse.kt
package com.bachld.android.data.dto.response.donhoandoan

import java.time.LocalDateTime

data class DonHoanDoAnResponse(
    val id: Long?,
    val sinhVienId: Long?,
    val trangThai: String?,           // HoanState (PENDING/APPROVED/REJECTED)
    val lyDo: String?,
    val minhChungUrl: String?,
    val requestedAt: LocalDateTime?,
    val decidedAt: LocalDateTime?,
    val nguoiPheDuyetId: Long?,
    val ghiChuQuyetDinh: String?
)
